### DISCUSSION TITLE
ERC20Mock import constructor not accepting any arguments

### DISCUSSION THREAD

**Discussion by Edacheclarus2** [Cyfrin/foundry-full-course-cu#765] [General]
I'm currently on the DecentralisedStableCoin  course and i'm facing errors trying to pass arguments to the new initialised ERC20Mock but getting error:
Even chatGpt says the arguments on the ERC20Mock are default but the course has it:
below are the code, original ERC20Mock(The one im importing) and error

MY CODE:
        vm.startBroadcast();
        MockV3Aggregator ethUsdPriceFeed = new MockV3Aggregator(
            DECIMALS,
            ETH_USD_PRICE
        );
        ERC20Mock wethMock = new ERC20Mock("WETH", "WETH", msg.sender,  1000e8);

        MockV3Aggregator btcUsdPriceFeed = new MockV3Aggregator(
            DECIMALS,
            BTC_USD_PRICE
        );
        ERC20Mock wbtcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);
        vm.stopBroadcast();
        
        ERROR
Error (6160): Wrong argument count for function call: 2 arguments given but expected 0.
  --> script/HelperConfig.s.sol:54:30:
   |
54 |         ERC20Mock wethMock = new ERC20Mock("WETH", 1000e8);
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6160): Wrong argument count for function call: 4 arguments given but expected 0.
  --> script/HelperConfig.s.sol:60:30:
   |
60 |         ERC20Mock wbtcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);

THE ERC20MOCK code from openzeppelin:
contract ERC20Mock is ERC20 {
    constructor() ERC20("ERC20Mock", "E20M") {}

    function mint(address account, uint256 amount) external {
        _mint(account, amount);
    }

    function burn(address account, uint256 amount) external {
        _burn(account, amount);
    }
}

Soeone help me out, please

**Comment by shikhar229169** [comment]
Actually in the latest versions of open zeppelin, they have removed all 4 arguments from constructor:
The latest version of ERC20Mock: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/mocks/token/ERC20Mock.sol
The previous version: 
```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.0;

import "../token/ERC20/ERC20.sol";

// mock class using ERC20
contract ERC20Mock is ERC20 {
    constructor(
        string memory name,
        string memory symbol,
        address initialAccount,
        uint256 initialBalance
    ) payable ERC20(name, symbol) {
        _mint(initialAccount, initialBalance);
    }

    function mint(address account, uint256 amount) public {
        _mint(account, amount);
    }

    function burn(address account, uint256 amount) public {
        _burn(account, amount);
    }

    function transferInternal(
        address from,
        address to,
        uint256 value
    ) public {
        _transfer(from, to, value);
    }

    function approveInternal(
        address owner,
        address spender,
        uint256 value
    ) public {
        _approve(owner, spender, value);
    }
}
```

So, you can just copy paste the above previous version into your ERC20Mock

**Reply by Edacheclarus2** [reply]
Thank you, man

Although I'm currently using ERC20Mock without the argument/Param.

**Comment by sunnyStefi** [comment]
I mention another solution [here](https://github.com/Cyfrin/foundry-defi-stablecoin-f23/issues/37#issuecomment-1742642838)

**Comment by iftikharuddin** [comment]
As mentioned in the course updates that in the latest version of **openzeppelin/openzeppelin-contracts** ERC20Mock.sol file has been updated with **0** constructor parameters

You should use **_forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit_** command to download the appropriate version synced with the video.

**Comment by iftikharuddin** [comment]
As mentioned in the course updates that in the latest version of **openzeppelin/openzeppelin-contracts** ERC20Mock.sol file has been updated with **0** constructor parameters

You should use **_forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit_** command to download the appropriate version synced with the video.
